### PR TITLE
cql-pytest: document how to write new cql-pytest tests

### PR DESCRIPTION
**Pull request by nyh**
Add to test/cql-pytest/README.md an explanation of the philosophy
of the cql-pytest test suite, and some guideliness on how to write
good tests in that framework.

### CODE REVIEW

File: test/cql-pytest/README.md [hunk 1]
```diff
@@ -42,3 +42,163 @@ Additional useful pytest options, especially useful for debugging tests:
...
+   come up as random draws of strings of length 10.
+   Another danger of randomized tests is that they tend to obscure what
+   is actually being tested: For example, a reviewer of the test may think
+   the empty string is included in the test, while actually it isn't.
+   Randomized "fuzz" testing has its benefits, but it is almost always the
+   wrong thing to do in the context of the cql-pytest framework. We should
+   probably have a separate framework (or at least separate files) for
+   these tests.
+
+7. **Write tests, not a test library**: Developers are often told that
```
Comment (alecco) [inline test/cql-pytest/README.md:142]: This is like, your opinion.

File: test/cql-pytest/README.md [hunk 2]
```diff
@@ -42,3 +42,163 @@ Additional useful pytest options, especially useful for debugging tests:
...
+   utility functions, or utility functions with many confusing parameters.
+   We've seen this happening in other frameworks such as dtest.
+   If you believe something _should_ be a utility function, start by
+   putting it inside the single test file that needs it - and only move
+   it to util.py if several test files can benefit from it.
+   At the time of this writing, cql-pytest has over 20,000 lines of test
+   code, and around of 500 lines of library code. Please keep this ratio.
+   We're writing a collection of tests - not a library.
+
+8. **Do not over-design**: Continuing the "we're writing tests, not a
```
Comment (alecco) [inline test/cql-pytest/README.md:164]: Again. Arguable. It's convenient to make easier to write tests (many/most developers don't want to in the first place). Also same rules of code duplication apply to test code as in non-test code. But this is just my opinion.

**Review by alecco — APPROVED** [review]
LGTM

File: test/cql-pytest/README.md (hunk 1 above)
Comment (denesb) [reply test/cql-pytest/README.md:142]: False code sharing is a real thing.

**Review by denesb — APPROVED** [review]

**Pull request closed without merging**